browser/scripts/ui-core.js

```diff
@@ -436,6 +462,120 @@ VizorUI.prototype.onKeyPress = function(e) {
 			break;
 	}
 
+	// keys for program-mode (noodles visible) only:
+	if (!E2.app.worldEditor.isActive()) {
+		switch(key) {
+			case uiKeys.gotoParentGraph:
+				if (E2.core.active_graph.parent_graph)
+					E2.app.onGraphSelected(E2.core.active_graph.parent_graph);
+				break;
+			case uiKeys.viewSource:
+			case 'shift+' + uiKeys.viewSource:
+			case 'alt+shift+' + uiKeys.viewSource:	// .fi
+				this.viewSource();
+				e.preventDefault();
+				break;
+			case uiKeys.moveSelectedNodesUp:
+				E2.app.executeNodeDrag(
+					E2.app.selectedNodes,
+					E2.app.selectedConnections,
+					0, -10);
+				e.preventDefault();
+				break;
+			case uiKeys.moveSelectedNodesDown:
+				E2.app.executeNodeDrag(
+					E2.app.selectedNodes,
+					E2.app.selectedConnections,
+					0, 10);
+				e.preventDefault();
+				break;
+			case uiKeys.moveSelectedNodesLeft:
+				E2.app.executeNodeDrag(
+					E2.app.selectedNodes,
+					E2.app.selectedConnections,
+					-10, 0);
+				e.preventDefault();
+				break;
+			case uiKeys.moveSelectedNodesRight:
+				E2.app.executeNodeDrag(
+					E2.app.selectedNodes,
+					E2.app.selectedConnections,
+					10, 0);
+				e.preventDefault();
+				break;
+		}
+
+		if (key >= uiKeys.number0 && key <= uiKeys.number9) { // 0-9
+			var numberHotKeys = [
+				'plugin:output_proxy', // 0
+				'plugin:input_proxy', // 1
+				'plugin:graph', // 2
+				'plugin:slider_float_generator', // 3
+				'plugin:const_float_generator', // 4
+				'plugin:float_display', // 5
+				'plugin:multiply_modulator', // 6
+				'preset:time_oscillate_between_2_values', // 7
+				'preset:image_show_image', // 8
+				'plugin:knob_float_generator' // 9
+			]
+
+			var item = numberHotKeys[key - uiKeys.number0]
```

Comment:
uhm.. there's other things like this, i take it back, best to handle as separate issue.